### DISCUSSION TITLE
BDD100k download on remote machine

### DISCUSSION THREAD

**Discussion by ArthDh** [bdd100k/bdd100k#150] [Q&A] [answered]
From my understanding, to download the data needs user to be logged into their account and then select the respective file to download. I however want to download the dataset on a remote machine, I was wondering if there was any detail that I am missing regarding the download that allows me to directly perform this action?

I tried creating a cURL request but I think it is being blocked due to authentication, any help would be appreciated!

**Comment by kariefury** [comment, marked as answer]
Hello, I just used wget {http://LoggedInURL/zipFileNameHere.zip} and replaced the URL in the curly brackets with the URL from the logged in data website. It downloaded without issues.

**Reply by ArthDh** [reply]
I see, I'll try that next time. Thank you!

**Reply by ukdsvl** [reply]
> Hello, I just used wget {[http://LoggedInURL/zipFileNameHere.zip}](http://LoggedInURL/zipFileNameHere.zip%7D) and replaced the URL in the curly brackets with the URL from the logged in data website. It downloaded without issues.

@kariefury I did try that, but the the URL that is appearing for me after logging in, is not meaningful in the first place. Hence, I am not able to download via wget. Can you please suggest how to make it work ?

**Reply by kariefury** [reply]
After you login, you can navigate to this URL : https://bdd-data.berkeley.edu/portal.html#download and scroll down a bit, there are all these buttons. I clicked on **"video parts"** and get a website looking like this: 
![image](https://user-images.githubusercontent.com/4248642/128413628-e7146324-ad0f-419d-8802-a884a49dbc8e.png)
It has a list of .zip files there.

Then right click and "copy URL" of one of those .zip files and use that for wget.

`wget http://dl.yf.io/bdd100k/video_parts/FILENAME.zip`

FILENAME is not the real name, I don't want to post the exact URL since it's for logged in download users. 

The filename's follow a predictable convention, so to grab the entire dataset (in video part's format), I just made a whole bunch of nearly identical wget requests.

**Reply by ukdsvl** [reply]
Firstly, many thanks for your prompt, detailed response. What you are referring to is "Video Parts" (as also mentioned in https://doc.bdd100k.com/download.html). I indeed manually scrolled down to the "100k Images" button which is relevant to me, and also manually downloaded the "Detection 2020 Labels". Additionally, I did separately come across "http://dl.yf.io/bdd100k/video_parts" and was hoping something similar to "100k images" would also be present, but couldn't find any. 

But, now that you have mentioned, I feel that I can indirectly make use of the smaller file "Detection 2020 Labels" and write a script to download the individual images from http://dl.yf.io/bdd100k/images/all_zip/ (guessing so ... shall post if i succeed).
